annotated.go

```diff
@@ -89,3 +90,263 @@ func (a Annotated) String() string {
 	}
 	return fmt.Sprintf("fx.Annotated{%v}", strings.Join(fields, ", "))
 }
+
+// annotations is used for building out the info needed to generate struct
+// with tags using reflection.
+type annotations struct {
```

Comment:
minor: This is now an internal representation of the annotated func. Consider renaming this to `annotatedFunc` or similar.